File: src/lib/registration-profile-complete.ts
/**
 * Единые критерии «регистрация завершена» для входа и базовой навигации.
 * Совпадают с валидацией форм в `register-profile-schema.ts`.
 */

import type { User } from "@/lib/types";

import {
  isNormalizedUsernameTokenAllowed,
  normalizeUsernameCandidate,
} from "@/lib/username-candidate";

const EMAIL_LIKE = /^[^\s@]+@[^\s@]+\.[^\s@]+$/u;

function normalizedUsername(raw: string): string {
  return normalizeUsernameCandidate(raw);
}

/**
 * @param u — документ `users/*` или снимок профиля из Auth-страницы / дашборда
 */
export function isRegistrationProfileComplete(
  u: Pick<User, "name" | "username" | "phone" | "email"> | null | undefined,
): boolean {
  if (!u) return false;
  const name = String(u.name ?? "").trim();
  if (name.length < 2) return false;
  const username = normalizedUsername(String(u.username ?? ""));
  if (!isNormalizedUsernameTokenAllowed(username)) return false;
  const email = String(u.email ?? "").trim();
  if (!email || !EMAIL_LIKE.test(email)) return false;
  return true;
}

File: mobile/packages/lighchat_firebase/lib/src/registration/registration_profile_complete.dart
import 'username_candidate.dart';

final _emailLike = RegExp(r'^[^\s@]+@[^\s@]+\.[^\s@]+$');

bool isRegistrationProfileComplete({
  required String? name,
  required String? username,
  required String? phone,
  required String? email,
}) {
  final n = (name ?? '').trim();
  if (n.length < 2) return false;
  final u = normalizeUsernameCandidate(username ?? '');
  if (!isNormalizedUsernameTokenAllowed(u)) return false;
  final e = (email ?? '').trim();
  if (e.isEmpty || !_emailLike.hasMatch(e)) return false;
  return true;
}
